### PR TITLE
sort nlu/story files before training

### LINKED ISSUE #5405: Model hashes are not equal when training from multiple nlu.md files
**Rasa version**:
1.8.1
**Python version**:
Python 3.7.0
**Operating system** (windows, osx, ...):
Ubuntu 19.10
**Issue**:
NLU models trained from multiple .md files have different md5sum even though the random seed was set. When training with the same data from one .md5 then models have the same hashes. 
Probably fix https://github.com/RasaHQ/rasa/pull/4898/commits/ad1dd2ccb36b6d9da522073cafa8d56bc19e9370#diff-b69ac23b9d0ad7d535a81c9f705d10e5 did not solved the issue. 

**Error (including full traceback)**:
```
Model 1 md5sums:
7898e02645b8bd479567355b7bd9dfcc  1/nlu/component_5_DIETClassifier.tf_model.data-00000-of-00001
d4c46fac513c3a4c63b693e17addfc2f  1/nlu/component_5_DIETClassifier.tf_model.index

Model 2 md5sums:
cd84319dddf5e3c4bcdcd4e89c49b2f3  2/nlu/component_5_DIETClassifier.tf_model.data-00000-of-00001
5297ccdca94857c71dc9a88a7b4b9744  2/nlu/component_5_DIETClassifier.tf_model.index
```

**Command or request that led to error**:
```
tar -xzf your_nlu_model.tar.gz
md5sum nlu/*
```

**Content of configuration file (config.yml)** (if relevant):
```yml
# Configuration for Rasa NLU.
# https://rasa.com/docs/rasa/nlu/components/
language: en
pipeline:
  - name: WhitespaceTokenizer
  - name: RegexFeaturizer
  - name: LexicalSyntacticFeaturizer
  - name: CountVectorsFeaturizer
  - name: CountVectorsFeaturizer
    analyzer: "char_wb"
    min_ngram: 1
    max_ngram: 4
  - name: DIETClassifier
    epochs: 100
    random_seed: 999
  - name: EntitySynonymMapper
  - name: ResponseSelector
    epochs: 100

# Configuration for Rasa Core.
# https://rasa.com/docs/rasa/core/policies/
policies:
  - name: MemoizationPolicy
  - name: TEDPolicy
    max_history: 5
    epochs: 100
  - name: MappingPolicy
```

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- model would always retrain if there's e.g. multiple files in a `data/nlu` directory - now that should be fixed.
- fixes #5405 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
hm ok tests are failing bc of types - is it fine that it's a list now or should i turn it into a set again?

File: rasa/data.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def get_core_nlu_files(
             story_files.update(new_story_files)
             nlu_data_files.update(new_nlu_data_files)
 
-    return story_files, nlu_data_files
+    return sorted(story_files), sorted(nlu_data_files)
```
Comment (wochinge) [inline rasa/data.py]: ideally that would also be covered by a test

Comment (akelad) [reply rasa/data.py]: honestly writing tests is not my forte 😅 there's no test for this function at all from what i can see. do you mind taking over?

Comment (akelad) [reply rasa/data.py]: actually, due to our wonderful testing file structure, i just didn't see the right tests. It's actulally implicitly covered by https://github.com/RasaHQ/rasa/blob/1.8.x/tests/core/test_data.py#L157 imo. 
I need to fix that test anyways

**Pull request merged**